diffusion-model-with-mnist-data.py

```diff
@@ -167,3 +165,46 @@ def forward(self, noisy_images, timestep):
 # # Evaluation
 
 
+# -
+
+# # Sampling
+
+# +
+def sample_image(image, beta, noise_predictor, simple_variance=False):
+    "sample image"
+    
+    image_size = image.shape
+    timesteps = beta.shape[0] - 1
+    alpha = 1 - beta
+    alpha_cum = torch.cumprod(alpha, dim=0)
+    if simple_variance:
+        variances = beta
+    else:
+        alpha_cum_t_minus_1 = torch.cat(torch.Tensor([0]), alpha_cum)
+        variances = (1-alpha_cum_t_minus_1)/(1-alpha_cum)
```

Comment:
Should not work; alpha_cum_t_minus_1 has one element more than alpha_cum. Fix it.